app/services/foreman/access_permissions.rb

```diff
@@ -19,7 +19,8 @@
     map.permission :edit_architectures,
                    :architectures => [:edit, :update],
                    :"api/v1/architectures" => [:update],
-                   :"api/v2/architectures" => [:update]
+                   :"api/v2/architectures" => [:update],
+                   :"api/v2/many_to_many" => [:create, :destroy]
```

Comment:
@ShimShtein, not sure that I follow. Your "shortest match" idea still needs to be based on defined permissions in `access_permissions.rb`. I don't think we want to have a bunch of extra permissions like `map.permission :edit_architectures_links_operatingsystems`. Just having `map.permission :edit_architectures` should be enough. If a user can `edit_architectures` then he should be able to add/remove operating systems for an architecture which is what I tried to do by adding `:"api/v2/many_to_many" => [:create, :destroy]` to the `edit_` permissions.